test/unit/test_runner_config.py

```diff
@@ -407,19 +407,31 @@ def test_prepare_with_ssh_key(open_fifo_write_mock):
 def test_wrap_args_with_ssh_agent_defaults():
     rc = RunnerConfig('/')
     res = rc.wrap_args_with_ssh_agent(['ansible-playbook', 'main.yaml'], '/tmp/sshkey')
-    assert res == ['ssh-agent', 'sh', '-c', 'ssh-add /tmp/sshkey && rm -f /tmp/sshkey && ansible-playbook main.yaml']
+    assert res == [
+        'ssh-agent',
+        'sh', '-c',
+        "trap 'rm -f /tmp/sshkey' EXIT && ssh-add /tmp/sshkey && rm -f /tmp/sshkey && ansible-playbook main.yaml"
```

Comment:
unrelated,

shred is a great util to zero out a file, then delete.

Reply:
TIL! Thanks @pabelanger. I will need to test that out and make sure it works for fifos that have already been read. If it does, that may be a safer solution here.